### ISSUE TITLE
False positive with `await-async-query` when followed by destructuring without declaration

### ISSUE THREAD

**Issue by siminino** [testing-library/eslint-plugin-testing-library#635]
### Have you read the Troubleshooting section?

Yes

### Plugin version

v5.6.0

### ESLint version

v7.32.0

### Node.js version

16.31.1

### package manager and version

yarn 1.22.19

### Operating system

macOS Monterey

### Bug description

The rule `await-async-query` identifies error when an awaited query is followed by a destructuring array without variable declaration and semicolons is disallowed.

I have this false positive when trying to fix `prefer-destructuring` (rule which we have enabled) on the next line.

```javascript
    // ...
    buttons = await screen.findAllByRole('button')
    ([removeButton] = buttons)
    userEvent.click(removeButton)
    // ...
```

### Steps to reproduce

1. Declare a variable;
2. Call test-library query which returns array (findAll*) with await;
3. Assign the value first returned value from de query to the previously declared variable using destructuring;

### Error output/screenshots

<img width="904" alt="Screenshot 2022-08-30 at 15 11 43" src="https://user-images.githubusercontent.com/632044/187445558-650b9457-af65-488c-ae9b-61575236aa62.png">

### ESLint configuration

module.exports = {
  parser: 'babel-eslint',
  rules: {
    'semi': ['error', 'never'],
    'prefer-destructuring': 'warn',
    'testing-library/await-async-query': 'error',
  },
  plugins: ['testing-library'],
}

### Rule(s) affected

- testing-library/await-async-query
- prefer-destructuring

### Do you want to submit a pull request to fix this bug?

No

**Comment by Belco90** [comment]
Hi @siminino! Thanks for reporting this problem, I was able to reproduce it locally, so I can confirm this is a bug.

This is affecting `testing-library/await-async-query` rule, but `prefer-destructuring` doesn't belong here.

**Comment by siminino** [comment]
Thanks @Belco90,
A quick workaround for those facing this issue is adding semicolon on the beginning of destructuring line.
Obs: I don't know if this workaround is going to work to any other rules combination.
```
    // ...
    buttons = await screen.findAllByRole('button')
    ;([removeButton] = buttons)
    userEvent.click(removeButton)
    // ...
```

**Comment by siminino** [comment]
Great explanation @sjarva. Thanks

Obs: The problem with the other workaround you suggested is that i can't reuse variables. This is needed in some cases.

**Comment by Belco90** [comment]
@sjarva Amazing explanation! I didn't know about this specific case for considering such code a call expression together with the previous line.

This made me check ESLint rules to prevent this, and I found the [`no-unexpected-multiline`](https://eslint.org/docs/latest/rules/no-unexpected-multiline) rule. It turns out this rule is catching an error in the provided example for this issue, confirming what @sjarva explained previously. [You can check this yourself in this ESLint playground](https://eslint.org/play/#eyJ0ZXh0IjoiLy8gLi4uXG5hc3luYyBmdW5jdGlvbiBleGFtcGxlV3JhcHBlcigpIHtcbiBidXR0b25zID0gYXdhaXQgc2NyZWVuLmZpbmRBbGxCeVJvbGUoJ2J1dHRvbicpXG4gKFtyZW1vdmVCdXR0b25dID0gYnV0dG9ucylcbiB1c2VyRXZlbnQuY2xpY2socmVtb3ZlQnV0dG9uKSBcbn1cbi8vIC4uLiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXX0sImVudiI6eyJlczYiOnRydWV9fX0=).

Since the code provided in this issue is considered a confusing multiline expressions reported by an ESLint built-in rule, I don't think this is an error in our plugin that should be fixed.

**Comment by siminino** [comment]
@Belco90 but don't you consider a proper detail for this kind of fail check in this rule?
While writing code now, is really hard to figure out what is happening by reading the fail message from this rule.

**Comment by Belco90** [comment]
> @Belco90 but don't you consider a proper detail for this kind of fail check in this rule? While writing code now, is really hard to figure out what is happening by reading the fail message from this rule.

Not for now, sorry. This is a really edge case that is caused by a different issue in the code, which is reported by another rule, so it's not worth it spending time in reporting such case.
